server/src/main/java/org/elasticsearch/action/admin/cluster/allocation/TransportGetAllocationStatsAction.java

```diff
@@ -88,10 +90,11 @@ protected void doExecute(Task task, Request request, ActionListener<Response> li
     protected void masterOperation(Task task, Request request, ClusterState state, ActionListener<Response> listener) throws Exception {
         listener.onResponse(
             new Response(
-                allocationStatsService.stats(),
-                featureService.clusterHasFeature(clusterService.state(), AllocationStatsFeatures.INCLUDE_DISK_THRESHOLD_SETTINGS)
-                    ? diskThresholdSettings
-                    : null
+                NodesStatsRequestParameters.Metric.ALLOCATIONS.containedIn(request.metrics()) ? allocationStatsService.stats() : Map.of(),
+                NodesStatsRequestParameters.Metric.FS.containedIn(request.metrics())
+                    && featureService.clusterHasFeature(clusterService.state(), AllocationStatsFeatures.INCLUDE_DISK_THRESHOLD_SETTINGS)
+                        ? diskThresholdSettings
+                        : null
```

Comment:
I am surprised to see FS related metrics in `cluster:monitor/allocation/stats`.
I assume it is tricky to rename it to something more generic because of bwc, but also why is it not possible to compute diskThresholdSettings locally?

Reply:
the settings which matter are the ones on the master, the node that's coordinating the stats request may have different local settings but these are not relevant